### ISSUE TITLE
Add search via plain HTTP request

### ISSUE THREAD

**Issue by serpent7776** [ocaml/opam2web#176]
Currently there's only search via AJAX. It would be nice to query packages via GET under e.g. `https://opam.ocaml.org/search/?q=foo` to search for `foo` package. This will also help fix #153.

**Comment by kanishka-work** [comment]
I don't believe this is currently technically feasible from the website, given its focus on static generation.

I assume opam has a server itself for serving packages. Might be better to move this issue to the codebase tracking features for that server

**Comment by serpent7776** [comment]
Should I open a new issue in some other repo? Where would that be?

**Comment by dra27** [comment]
It’s fine here, it’s just not terribly likely to happen.

The search isn’t currently done using AJAX, it’s simply done on the client. opam2web is designed to generate a static site, so if your intention was to be able to have this without JavaScript, that’s pretty much never going to happen.

That said, it would be possible to have the page read the query string and do the filtering.

Please note (from an open source perspective) that “it would be nice if” is not the most compelling reason for others to pick the issue up - “I can’t do x because opam2web doesn’t recognise query strings” is more convincing 🙂

**Comment by serpent7776** [comment]
My intention was to be able to use it with search shortcuts in the browser, so that I can type e.g. `opam foo` in address bar and opam site will be opened with the `foo` query used to filter entries. This requires a `GET` parameter to be present.
I guess that reading the query on page load would be enough for that.
